src/rust/engine/workunit_store/src/metrics.rs

```diff
@@ -48,6 +48,8 @@ pub enum Metric {
   RemoteExecutionRPCWaitExecution,
   RemoteExecutionSuccess,
   RemoteExecutionTimeouts,
+  SpeculationLocalExecutionCompletedFirst,
+  SpeculationRemoteCacheCompletedFirst,
```

Comment:
Nit: Namespacing wise, would put RemoteCache on the front.

Reply:
In terms of naming, I have been putting the "subsystem" into the prefix of the name. Now `Speculation` may not be the right subsystem name, but for consistency with the existing metrics, whatever is chosen as the subsystem name should come first in the metric name.

Reply:
`RemoteCacheSpeculationLocalCompletedFirst`
`RemoteCacheSpeculationRemoteCompletedFirst`

Reply:
Tom had suggested this thinking that we might want to eventually add `SpeculationRemoteExecutionCompletedFirst`. That is, the speculation is the interesting part, rather than the remote cache. But I'm fine with either approach.

Reply:
Yeah I had viewed "speculation" as the subsystem.

Reply:
Keeping in mind that this remains a "nit", but:

This patch will be the second place in the codebase to speculate right now, and the third will come along with remote-execution ([see](https://docs.google.com/document/d/1hS0n6vp-hBxy4Xo-q9QA_elofzQtOubjfN4nSeIgp90/edit#)). Speculation is an implementation detail of multiple components, rather than a component in-and-of-itself.